exec/java-exec/src/main/java/org/apache/drill/exec/store/dfs/FileSelection.java

```diff
@@ -118,13 +133,34 @@ public boolean apply(@Nullable FileStatus status) {
       }
     }));
 
-    return create(nonDirectories, null, selectionRoot);
+    final FileSelection fileSel = create(nonDirectories, null, selectionRoot);
```

Comment:
I did not change the existing create() call .. it was passing null for the list of files.  I think it is ok because FileStatus contains the Path which reflects the file name.  It also seems that if either the list of files is null or the list of statuses is null, the functions getFiles() or getStatuses() can compute them on the first invocation and save for future use.